projects/common/src/navigation/navigation.service.ts

```diff
@@ -105,13 +106,15 @@ export class NavigationService {
   public navigateWithinApp(
     path: NavigationPath,
     relativeTo?: ActivatedRoute,
-    preserveParameters?: string[]
+    preserveParameters?: string[],
+    skipLocationChange?: boolean
```

Comment:
this is a shortcut method, instead of modifying this with more optional parameters, can you use `navigate` instead?

Reply:
What will be better - create new method similar to navigateWithinApp in navigation service (maybe it'll be helpful in future or it's a rare case?) or just directly use navigate method inside navigation-list component without any shortcut?

Reply:
I would leave navigationService untouched, and call via:
```typescript
  this.navigationService.navigate({
      navType: NavigationParamsType.InApp,
      path: item.matchPaths[0],
      replaceCurrentHistory: item.skipLocationChange,
      relativeTo: this.activatedRoute
  });
```

Replacing the history, when changing something other than a query param (which is already supported) should be a rare case. I'm not convinced we need it here, but I definitely don't want to encourage it for future use cases.

Reply:
(you beat me to this with the exact change 😄 )